### PR TITLE
[LLM] update llama parameters

### PR DESCRIPTION
**Pull request by linjieccc**
### PR types

Others
### PR changes

Models
### Description

- update llama parameters

### CODE REVIEW

**Review by wawltor — DISMISSED** [review]
LGTM

File: examples/language_model/llama/finetune_generation.py [hunk 1]
```diff
@@ -90,27 +89,36 @@ def main():
...
     model = AutoModelForCausalLM.from_pretrained(
         model_args.model_name_or_path,
         load_state_as_np=True,
         low_cpu_mem_usage=True,
-        # dtype="float16",  # todo enable set dtype to avoid additional mem usage
+        dtype=dtype,  # todo enable set dtype to avoid additional mem usage
         tensor_parallel_degree=training_args.tensor_parallel_degree,
         tensor_parallel_rank=training_args.tensor_parallel_rank,
         use_recompute=True,
+        use_pure_fp16=training_args.fp16,
```
Comment (wawltor) [inline examples/language_model/llama/finetune_generation.py]: 删除use_pure_fp16 flag

File: paddlenlp/transformers/llama/modeling.py [hunk 2]
```diff
@@ -648,22 +667,42 @@ def forward(self, prediction_scores, masked_lm_labels):
         masked_lm_loss = self.loss_func(prediction_scores, masked_lm_labels.unsqueeze(2))
         with paddle.amp.auto_cast(False):
             masked_lm_loss = masked_lm_loss.astype("float32")
-            masked_lm_loss = masked_lm_loss[masked_lm_labels != -100]
+            masked_lm_loss = masked_lm_loss[masked_lm_labels != 0]
```
Comment (wawltor) [inline paddlenlp/transformers/llama/modeling.py:684]: padding index，而不是hard code

File: paddlenlp/transformers/llama/modeling.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 import numpy as np
 import paddle
 import paddle.nn.functional as F
-from paddle import nn
+from paddle import Tensor, nn
```
Comment (blldd) [inline paddlenlp/transformers/llama/modeling.py:24]: 您好，请问这个对应的是paddle的哪个版本呢，目前我安装的是2.4.2，没有Tensor这个函数。

Comment (linjieccc) [reply paddlenlp/transformers/llama/modeling.py:24]: @blldd 你好，需要安装下develop版本的paddle

https://www.paddlepaddle.org.cn/install/quick?docurl=/documentation/docs/zh/develop/install/pip/linux-pip.html

**Review by wawltor — COMMENTED** [review]
LGTM

**Review by wawltor — APPROVED** [review]
LGTM
